solution/index.js

```diff
@@ -0,0 +1,316 @@
+}
+//find the ul we want to move the task to
+function findKeyList(event){
+    if(event.altKey && event.key === "1" )
+    {
+        moveTask(toDoListEl , toDoStorage);
+    }
+    else if(event.altKey && event.key === "2" )
+    {
+        moveTask(inProgressListEl , inProgressStorage);
+    }
+    else if(event.altKey && event.key === "3" )
+    {
+        moveTask(doneListEl , doneStorage);
+    }
+}
+// delete the task from his old ul and add it to the new one
+function moveTask(domList , storageList ){
+    let oldUl = elementThatMove.parentElement.className; // the old ul of the task that we want to move
+    if(oldUl === "to-do-tasks")
+    {
+        let i = toDoStorage.indexOf(elementThatMove.innerText);
+        toDoStorage.splice(i,1);
+    }
+    else if(oldUl === "in-progress-tasks")
+    {
+        let i = inProgressStorage.indexOf(elementThatMove.innerText);
+        inProgressStorage.splice(i,1);
+    }
+    else if(oldUl === "done-tasks")
+    {
+        let i = doneStorage.indexOf(elementThatMove.innerText);
+        doneStorage.splice(i,1);
+    }
+    domList.insertBefore(elementThatMove, domList.firstChild); // move the task in the dom
+    storageList.unshift(elementThatMove.innerText); // move the task in the local storage
+    localStorage.setItem("tasks" ,JSON.stringify(localStorageObj));
+}
+
+
+// show only the tasks with the same text of the search input
+const searchEl = document.getElementById("search");
+searchEl.addEventListener("keyup" , searchTasks);
+function searchTasks(){
+    let allLi = document.getElementsByClassName("task");
+    for(let list of allLi){
+        if(!list.innerText.toLowerCase().includes(searchEl.value.toLowerCase()))
+        {
+        list.style.display = "none";
+        }else
+        {
+            list.style.display = "";
+        }
+    }
+}
+
+let dragged;
+
+document.addEventListener("dragstart", function( event ) {
+    dragged = event.target; // store a ref. on the dragged elem
+    event.target.style.opacity = .5; // make it half transparent
+}, false);
+
+document.addEventListener("dragend", function( event ) {
+    event.target.style.opacity = ""; // reset the transparency
+}, false);
+
+ /* events fired on the drop targets */
+ document.addEventListener("dragover", function( event ) {
+    // prevent default to allow drop
+    event.preventDefault();
+}, false);
+
+document.addEventListener("dragenter", function( event ) {
+    // highlight potential drop target when the draggable element enters it
+    if ( event.target.className === "dropzone" ) {
+        event.target.style.background = "purple";
+    }
+}, false);
+
+document.addEventListener("dragleave", function( event ) {
+    // reset background of potential drop target when the draggable element leaves it
+    if ( event.target.className === "dropzone" ) {
+        event.target.style.background = "";
+    }
+}, false);
+
+document.addEventListener("drop", function( event ) {
```

Comment:
For the most part, a broad use of if else represents that more accurate logical thinking could have been used for the exercise.